pkg/clusteroperator/list.go

```diff
@@ -0,0 +1,121 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// List returns clusterOperators inventory.
+func List(apiClient *clients.Settings) ([]*Builder, error) {
+	glog.V(100).Info("Listing all clusterOperators")
+
+	coList, err := apiClient.ClusterOperators().List(context.Background(), metaV1.ListOptions{})
+
+	if err != nil {
+		glog.V(100).Infof("Failed to list clusterOperators due to %s", err.Error())
+
+		return nil, err
+	}
+
+	var coObjects []*Builder
+
+	for _, clusterOperator := range coList.Items {
+		copiedCo := clusterOperator
+		coBuilder := &Builder{
+			apiClient:  apiClient,
+			Object:     &copiedCo,
+			Definition: &copiedCo,
+		}
+
+		coObjects = append(coObjects, coBuilder)
+	}
+
+	return coObjects, nil
+}
+
+// WaitForAllClusteroperatorsAvailable wait for all clusterOperators are in available state.
+func WaitForAllClusteroperatorsAvailable(apiClient *clients.Settings, timeout time.Duration) (bool, error) {
+	glog.V(100).Info("Waiting for all clusterOperators to be in available state")
+
+	coList, err := List(apiClient)
+	if err != nil {
+		glog.V(100).Infof("Failed to list all clusterOperators due to %s", err.Error())
+
+		return false, err
+	}
+
+	// Wait 5 secs in each iteration before condition function () returns true or errors or times out.
+	err = wait.PollImmediate(fiveScds, timeout, func() (bool, error) {
+
+		// iterate through the clusterOperators in the list.
+		for _, clusteroperator := range coList {
+			if !clusteroperator.IsAvailable() {
+				glog.V(100).Infof("The %s clusterOperator is not available",
+					clusteroperator.Object.Name)
+
+				return false, nil
+			}
+		}
+
+		return true, nil
+	})
+
+	if err == nil {
+		glog.V(100).Infof("All clusterOperators were found available before timeout: %v",
+			timeout)
+
+		return true, nil
+	}
+
+	// Here err is "timed out waiting for the condition"
+	glog.V(100).Infof("Not all clusterOperators were found available before timeout: %v",
+		timeout)
+
+	return false, err
+}
+
+// WaitForAllClusteroperatorsStopProgressing check that all pods in namespace that match options are in running state.
```

Comment:
I don't see where we check for pods/namespace here?
Had to look what List does (hence the suggestion to rename List to ClusterOperatorList)

Reply:
there is no namespace for clusteroperators.
why we need to check pods name, we are working with co, not pods.
it will be not renamed, it was already renamed to the List, per Nikita request.

Reply:
The comment you have reads:
"// WaitForAllClusteroperatorsStopProgressing check that all pods in namespace that match options are in running state.
"
So you tell me why.

@elenagerman I'd suggest refraining from sentences like "this will not happen".
"it was already renamed to the List, per Nikita request" is sufficient.

Reply:
I see, just a left-over from the copy/paste. Fixed now, thank you!

Reply:
@achuzhoy I didn't find where I used "this will not happened", but only "it will be not renamed" with an explanation why. But if you insist I remove it.